frontend/src/app/main/data/workspace.cljs

```diff
@@ -1638,13 +1638,14 @@
                   process-shape
                   (fn [_ shape]
                     (-> shape
-                        (assoc :frame-id frame-id)
-                        (assoc :parent-id parent-id)
+                        (assoc :frame-id frame-id :parent-id parent-id)
 
                         ;; if foreign instance, detach the shape
                         (cond-> (foreign-instance? shape paste-objects state)
-                          (dissoc :component-id :component-file :component-root?
-                                  :remote-synced? :shape-ref :touched))
+                          (->
+                             (assoc :foreign-component-root? (:component-root? shape))
```

Comment:
Suggestion: rename this to `:saved-component-root?` and add a comment saying this is used later, if the paste needs to create a new component.

Reply:
Ok, changed